### DISCUSSION TITLE
question: how do I ignore specific types of codeblocks?

### DISCUSSION THREAD

**Discussion by clubanderson** [facelessuser/pyspelling#175] [Q&A]
In our project we are scanning all .md files.  Those files contain blocks of code that are to be executed, and other blocks that are not intended to be executed - these blocks are just output from the commands.  The output blocks contain UIDs typically.  These fail a spellcheck of course.  I want to spellcheck codeblocks that have commands to run which are inside:

```shell
command-to-run.sh
```

and ignore blocks that have output:

``` { .bash }
output 477fdjjh3-dskjd-jkjkjsd
```

what do you suggest I use for ignoring this type of input to pyspelling.  Both blocks are wrapped in <code></code> when converted to html, but I ignore the generated files.

example of what we are spellchecking is https://docs.kubestellar.io/main/Coding%20Milestones/PoC2023q1/example1/

Thank you

**Comment by facelessuser** [comment]
I am assuming you are using the HTML filter. If you are, you can use the `ignores` option: https://facelessuser.github.io/pyspelling/filters/html/#options.

The document mentions that the CSS selector lib that is used is Soup Sieve (which I also maintain). You may find the custom pseudo-class [`:-soup-contains()`](https://facelessuser.github.io/soupsieve/selectors/pseudo-classes/#:-soup-contains) helpful.

**Reply by clubanderson** [reply]
But the use of ignores matches HTML, not the md, right? I am scanning the md files, not the resulting HTML.

**Reply by facelessuser** [reply]
I guess I usually use the [Markdown filter](https://github.com/facelessuser/coloraide/blob/main/.pyspelling.yml#L45) to convert the content to HTML and chain the HTML filter right after. An example here: https://github.com/facelessuser/coloraide/blob/main/.pyspelling.yml#L45. That is kind of a one-off thing to parse the readme. The parser being used is Python Markdown, assuming that parser is sufficient for your needs, this may be a possible approach. All of my documentation is built with Python Markdown, so I'm used to using that parser.

But really, for full documentation (not single Markdown files), what I usually do is build it and then run the spellchecker on the final HTML output. It is much easier to filter HTML than it is to filter Markdown at the level of granularity that you'd like. 

The [context filter](https://facelessuser.github.io/pyspelling/filters/context/) can be useful, but I'm not sure that your case really fits its strengths though.

**Reply by facelessuser** [reply]
It does seem you are using MkDocs Material, based on the link you provided. I use it as well. All I do is this: 

- Action: https://github.com/facelessuser/pyspelling/blob/master/.github/workflows/build.yml#L133
- Tox config: (though you could probably do this directly in the action if required): https://github.com/facelessuser/pyspelling/blob/master/tox.ini#L16
- Pyspelling config: https://github.com/facelessuser/pyspelling/blob/master/.pyspelling.yml#L4

I realize your setup may not be exactly like mine, but maybe this sparks some ideas of how you could do something similar within whatever framework you operate in. The main point is HTML is way easier to write rules for than Markdown, which is why I don't operate directly in Markdown.

**Comment by clubanderson** [comment]
Thanks @facelessuser.  I was looking for something a bit simpler.  Maybe your answer will make more sense to me when I have a bit more experience re: pyspelling

**Reply by facelessuser** [reply]
Understood, unfortunately, what you are asking for is somewhat complex. A Markdown environment requires a Markdown parser. As you are using MkDocs, you can potentially use custom Markdown extensions or MkDocs plugins making standard Markdown even more difficult to parse. The easiest way is to handle it is as HTML.

You could implement things more simply. I realize you may not be a Python developer so you may not have heard of things like tox. You don't actually need tox. Really, all you need to do is build your docs in the action and then run pyspelling on the HTML output. That's it. How you do that doesn't matter as much. The process can be done completely in the github action (plus pyspelling config).

**Comment by clubanderson** [comment]
tbh, what I would like to do is look for 

```
<!--pause-spellcheck-->
123232320u444uop-fjdkfjkd-hrejehrjeh
<!--resume-spellcheck-->

```
and ignore everything in that block

**Reply by facelessuser** [reply]
And you'd like to just process the whole file in plain text, not Markdown? If so, sure, you could do that with a [context filter](https://facelessuser.github.io/pyspelling/filters/context/). You may find delimiting your spellchecking this way limited in other ways, but maybe not. I'm not here to convince you of doing it one specific way, you are certainly free to explore alternate paths and do what works best for you.

**Comment by clubanderson** [comment]
ah - now that's what I was looking for, I think.  That's very helpful.  Let me give it a shot - thank you.
